### DISCUSSION TITLE
Filament V3 conditional error modal

### DISCUSSION THREAD

**Discussion by clinvest** [filamentphp/filament#9580] [Help] [answered]
Hi all,

I use Filament V3

How would you do to open a "conditional  error modal" (for example using the requiresConfirmation but without the submit button) on certain a condition.

If condition is false : open the error modal (with only the cancel button)
If condition is true : open the form modal

My code is getting pretty complex if I use inline functions for that. Just for example so you can understand what I'm trying to do (it's just to explain, obviously not working like that).

```
Action::make('resetStars')
    ->icon('heroicon-m-x-mark')
    ->color('danger')
    ->requiresConfirmation() // if $condition is false, show the error modal
    ->modalHeading('Send estimate') // if $condition is false, show the error modal
    ->modalDescription('Condition is false so you cannot send your estimate') // if $condition is false, show the error modal
    ->modalSubmitAction(false) // if $condition is false, show the error modal
    ->modalHidden(fn (): bool => ...) // if $condition is true, hide the error modal
    ->form([...]) // if $condition is true, show the form modal
    ->action(function () {
        ...
    })
```

Should I do a custom livewire component?

Thank you in advance,

**Comment by leandrocfe** [comment]
Why not [notifications](https://filamentphp.com/docs/3.x/notifications/sending-notifications#overview) ?

**Comment by coolsam726** [comment, marked as answer]
[Check this in the Docs](https://filamentphp.com/docs/3.x/support/blade-components/modal#controlling-a-modal-from-javascript)

**Reply by clinvest** [reply]
That's a way to do it, I'll consider this. Thank you!

**Comment by Umar-Farooq-Shafi** [comment]
The modal is still opening even if I set `requiresConfirmation(false)`.

<img width="888" alt="Screenshot 2024-03-19 at 1 56 37 PM" src="https://github.com/filamentphp/filament/assets/62417503/151c520e-88a6-4248-94e6-fe6cb74a442a">

**Reply by leandrocfe** [reply]
Open a new topic and share the code you are doing

**Reply by Umar-Farooq-Shafi** [reply]
@leandrocfe I fixed it by setting the `modalHidden` function. thanks
